lighthouse-core/test/gather/gatherers/console-test.js

```diff
@@ -0,0 +1,345 @@
+        args: [
+          {
+            type: 'string',
+            value: 'This is a warning!',
+          },
+        ],
+        executionContextId: 4,
+        timestamp: 1605300392523.653,
+        stackTrace: {
+          callFrames: [
+            {
+              url: 'http://www.example.com/fancybox.js',
+              lineNumber: 28,
+              columnNumber: 20,
+            },
+          ],
+        },
+      };
+
+    const driver = new MockDriver();
+    const options = {driver};
+
+    await consoleGatherer.beforePass(options);
+    driver.fireForTest('Runtime.consoleAPICalled', consoleWarnEvent);
+
+    const artifact = await consoleGatherer.afterPass(options);
+
+    assert.equal(artifact.length, 1);
+    assert.equal(artifact[0].source, 'console.warn');
+    assert.equal(artifact[0].level, 'warning');
+    assert.equal(artifact[0].text, 'This is a warning!');
+    assert.equal(artifact[0].url, 'http://www.example.com/fancybox.js');
+    assert.equal(artifact[0].lineNumber, 28);
+    assert.equal(artifact[0].columnNumber, 20);
+  });
+
+  it('captures console.warn calls with non-string args', async () => {
+    const consoleGatherer = new ConsoleGatherer();
+    const consoleWarnEvent =
+      {
+        type: 'warning',
+        args: [
+          {type: 'string', value: 'Testing'},
+          // Not JSON (the window)
+          {
+            type: 'object',
+            className: 'Window',
+            description: 'Window',
+            objectId: '{"injectedScriptId":4,"id":1}',
+            preview: [
+              {
+                type: 'object',
+                description: 'Window',
+                overflow: true,
+                properties: [
+                  {name: 'window', type: 'object', value: 'Window'},
+                  {name: 'self', type: 'object', value: 'Window'},
+                  {
+                    name: 'document',
+                    type: 'object',
+                    value: '#document',
+                    subtype: 'node',
+                  },
+                  {name: 'name', type: 'string', value: ''},
+                  {name: 'location', type: 'object', value: 'Location'},
+                ],
+              },
+            ],
+          },
+          // JSON: {isJson: true}
+          {
+            type: 'object',
+            className: 'Object',
+            description: 'Object',
+            objectId: '{"injectedScriptId":4,"id":2}',
+            preview: {
+              type: 'object',
+              description: 'Object',
+              overflow: false,
+              properties: [{name: 'json', type: 'boolean', value: 'true'}],
+            },
+          },
+          // A native function: console.log
+          {
+            type: 'function',
+            className: 'Function',
+            description: 'function log() { [native code] }',
+            objectId: '{"injectedScriptId":4,"id":3}',
+          },
+          // A defined function
```

Comment:
want to add a falsy value like `0` to test the primitive usage?

Reply:
Done.